scala/lambdas/ingest-mapper/src/main/scala/uk/gov/nationalarchives/ingestmapper/MetadataService.scala

```diff
@@ -88,7 +88,26 @@ class MetadataService(s3: DAS3Client[IO], discoveryService: DiscoveryService) {
           Stream.evals {
             topLevelIdsToDepartmentSeries.map { itemToDepartmentSeries =>
               val parentPaths = getParentPaths(json, itemToDepartmentSeries)
-              val updatedJson = jsonArr.map { metadataEntry =>
+              val childCountMap = parentPaths.groupBy(_._2).view.mapValues(_.size).toMap
+
+              def addChildCount(obj: Obj) = {
+                val values = obj.value.toMap
+                val id = values("id").str
+                val parentPath = values.get("parentPath").map(parentPath => s"${parentPath.str}/").getOrElse("")
+                val objId = s"$parentPath$id"
+                val childCount = childCountMap.getOrElse(objId, 1)
+                Obj.from(values + ("childCount" -> Num(childCount)))
+              }
+
+              val departmentSeriesObjects = itemToDepartmentSeries.values.toList
+                .distinctBy(item => item.potentialSeriesItem.map(obj => obj("name")))
```

Comment:
How come this is distinct by `name` and not `id`?

Reply:
The id is a random UUID so distinct by `id` won't deduplicate anything.